### PR TITLE
Remove fundraising store

### PR DESCRIPTION
**Pull request by Abban**
https://phabricator.wikimedia.org/T247744

Related pull requests:

- https://github.com/wmde/fundraising-donations/pull/85
- https://github.com/wmde/fundraising-memberships/pull/41

### CODE REVIEW

File: src/DataAccess/DoctrineEntities/Subscription.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	/**
+	 * @var DateTime
+	 * @Gedmo\Timestampable(on="create")
+	 * @ORM\Column(type="datetime")
+	 */
+	private $createdAt;
+
+	private const STATUS_NEW = 0;
+	private const STATUS_CONFIRMED = 1;
+	private const STATUS_MODERATION = 2;
```
Comment (gbirke) [inline src/DataAccess/DoctrineEntities/Subscription.php:86]: We can drop this status and all moderation-related code. Moderation was only needed for questionable addresses ("Dr. Fakey McFakeface")

Comment (Abban) [reply src/DataAccess/DoctrineEntities/Subscription.php:86]: Is `STATUS_CONFIRMED` used for email validation?

Comment (gbirke) [reply src/DataAccess/DoctrineEntities/Subscription.php:86]: yes, see https://github.com/wmde/fundraising-subscriptions/blob/master/src/UseCases/ConfirmSubscription/ConfirmSubscriptionUseCase.php

File: src/SubscriptionContextFactory.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+		$this->connection = null;
+		$this->entityManager = null;
+	}
+
+	public function getEntityManager(): EntityManager {
+		if ( $this->entityManager === null ) {
+			AnnotationRegistry::registerLoader( 'class_exists' );
+
+			$driver = new XmlDriver( self::DOCTRINE_CLASS_MAPPING_DIRECTORY );
```
Comment (gbirke) [inline src/SubscriptionContextFactory.php:39]: Please move the creation of the XML driver into a public `newMappingDriver` method and move the entity manager creation to test.

File: src/SubscriptionContextFactory.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\ORM\Mapping\Driver\XmlDriver;
+use Gedmo\Timestampable\TimestampableListener;
+
+class SubscriptionContextFactory {
+
+	private const DOCTRINE_CLASS_MAPPING_DIRECTORY = __DIR__ . '/../config/DoctrineClassMapping';
+
```
Comment (gbirke) [inline src/SubscriptionContextFactory.php:15]: We still need the public constant `ENTITY_NAMESPACE` for the chained mapping driver in the FF

**Comment by gbirke** [comment]
With the removal of of the address, you can also trim down the `SubscriptionRequest` class. Including removing its test and the `wikilogin` property..

File: src/Validation/SubscriptionValidator.php [hunk 4]
```diff
@@ -22,13 +22,12 @@ class SubscriptionValidator {
...
-	private $mailValidator;
-	private $duplicateValidator;
-	private $textPolicyValidator;
-	private $textPolicyViolations;
-	private $titleValidator;
+	private EmailValidator $mailValidator;
+	private SubscriptionDuplicateValidator $duplicateValidator;
+	private TextPolicyValidator $textPolicyValidator;
+	private array $textPolicyViolations;
+	private AllowedValuesValidator $titleValidator;
```
Comment (gbirke) [inline src/Validation/SubscriptionValidator.php:30]: You can also drop the the `$textPolicyValidator`, `$titleValidator` and `$textPolicyViolations`, I'll take care of removing those in the frontend.

**Pull request merged**